### PR TITLE
Refactoring happens

### PR DESCRIPTION
**Pull request by ndushay**
Some refactoring I did while working on #434 (improve audit timestamps).

Should be easy to review commit by commit.

### CODE REVIEW

File: app/services/preserved_object_handler.rb [hunk 1]
```diff
@@ -71,10 +71,7 @@ def check_existence
           # FIXME: what if there is more than one associated pres_copy?
           pres_copy = PreservedCopy.find_by!(preserved_object: pres_object, endpoint: endpoint) if pres_object
 
-          if pres_copy.version != pres_object.current_version
-            handler_results.add_result(PreservedObjectHandlerResults::PC_PO_VERSION_MISMATCH, { pc_version: pres_copy.version, po_version: pres_object.current_version })
-            raise ActiveRecord::Rollback, "PO current_version #{pres_object.current_version} != PC version #{pres_copy.version}"
-          end
+          raise_rollback_if_pc_po_version_mismatch(pres_copy.version, pres_object.current_version)
```
Comment (SaravShah) [inline app/services/preserved_object_handler.rb]: 👍

File: app/services/preserved_object_handler_results.rb [hunk 2]
```diff
@@ -68,7 +68,7 @@ def self.logger_severity_level(result_code)
 
   def initialize(druid, incoming_version, incoming_size, endpoint)
     @incoming_version = incoming_version
-    @msg_prefix = "PreservedObjectHandler(#{druid}, #{incoming_version}, #{incoming_size}, #{endpoint})"
+    @msg_prefix = "PreservedObjectHandler(#{druid}, #{incoming_version}, #{incoming_size}, #{endpoint.endpoint_name if endpoint})"
```
Comment (SaravShah) [inline app/services/preserved_object_handler_results.rb]: 👍

**Comment by ndushay** [comment]
I guess one question I forgot to ask was:  is endpoint_name sufficient?  (I hope so) ... or should we use storage_location instead?  or both?

**Comment by tallenaz** [comment]
LGTM. IMO, `endpoint_name` is sufficient, and has the benefit of being shorter.

**Pull request merged**